tests/fe/fe_enriched_color_07.cc

```diff
@@ -1339,12 +1339,15 @@ LaplaceProblem<dim>::build_fe_space()
   pcout << "...building fe space" << std::endl;
 
   make_enrichment_functions();
-  static ColorEnriched::Helper<dim> fe_space(fe_base,
-                                             fe_enriched,
-                                             vec_predicates,
-                                             vec_enrichments);
+
+  static std::unique_ptr<ColorEnriched::Helper<dim>> fe_space;
+  fe_space = std::make_unique<ColorEnriched::Helper<dim>>(fe_base,
+                                                          fe_enriched,
+                                                          vec_predicates,
+                                                          vec_enrichments);
+
```

Comment:
This has no effect at all. In one case, the object is destroyed at the end of the program. In the other case, the pointer is destroyed at the end of the program, which in turn causes the destruction of the object. There is no difference.

What happens is that the `ColorEnriched::Helper` object keeps a reference to some of its arguments, which die earlier. This is the cause of your failure. You have to make sure that this object lives not as long as the arguments, or the arguments live longer than the object.

Reply:
@bangerth The bug we found with `-fsanitize=adress` is that the original `static` storage initialized the `ColorEnriched::Helper` function only once, but we call the setup function a second time (for another 2D test). Then the pointers to `fe_base` and `fe_enriched` are invalid because these objects changed.

This fix sets up a unique pointer that gets reinitialized for the second run, which fixes the use after scope issue that `-fsanitize=address` reported.

It is correct that this `static std::unique_ptr` isn't elegant. But it fixes the problem with minimal changes.

Reply:
I see now. Bizarre. Good sleuthing!